### PR TITLE
update envoy to f958d39fcd

### PR DESCRIPTION
**Pull request by junr03**
Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

**Comment by junr03** [comment]
going to merge as is and open up a high priority issue to get a current diff on the size regression

File: library/common/BUILD [hunk 1]
```diff
@@ -35,7 +35,15 @@ envoy_cc_library(
...
-        "@envoy//source/exe:envoy_common_lib",
+        "@envoy//source/exe:main_common_lib",
+    ],
+)
+
+envoy_cc_library(
+    name = "envoy_mobile_main_common_lib_stamped",
+    repository = "@envoy",
+    deps = [
+        ":envoy_mobile_main_common_lib",
```
Comment (rebello95) [inline library/common/BUILD:46]: Why is this now required?

File: library/common/config_template.cc [hunk 2]
```diff
@@ -268,4 +268,11 @@ stats_flush_interval: {{ stats_flush_interval_seconds }}s
     app_id : {{ app_id }}
     app_version : {{ app_version }}
     os: {{ device_os }}
+# Needed due to warning in https://github.com/envoyproxy/envoy/blob/6eb7e642d33f5a55b63c367188f09819925fca34/source/server/server.cc#L546
```
Comment (rebello95) [inline library/common/config_template.cc:271]: Is there an issue for this or intended to be permanent?

File: library/common/BUILD (hunk 1 above)
Comment (junr03) [reply library/common/BUILD:46]: https://github.com/envoyproxy/envoy/pull/12171 split things for tests, so our targets here were either: a) not having symbols, or having duplicate symbols. This new setup lets us use the new upstream system correctly.

File: library/common/config_template.cc (hunk 2 above)
Comment (junr03) [reply library/common/config_template.cc:271]: Unfortunately from the TODOs in that code the only way to set that configuration is in runtime, and the only (for now) way we can set runtime in envoy mobile is this way

**Pull request merged**